packages/backend/src/peripherals/database/shared/BaseRepository.ts

```diff
@@ -20,9 +20,25 @@ type FindMethod<A extends unknown[], T> = (...args: A) => Promise<T | undefined>
 
 type DeleteMethod<A extends unknown[]> = (...args: A) => Promise<number>
 
-type SaveMethod<T> = (record: T) => Promise<boolean>
+type Keys<T, U> = Extract<keyof T, U>
+type Match<T, U> = T extends U ? T : never
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export type CheckConvention<T extends BaseRepository> = {
+  [K in Keys<T, `add${string}`>]: Match<T[K], AddMethod<any, any>>
+} & {
+  [K in Keys<T, `addMany${string}`>]: Match<T[K], AddManyMethod<any, any>>
+} & {
+  [K in Keys<T, `find${string}`>]: Match<T[K], FindMethod<any[], any>>
+} & {
+  [K in Keys<T, `get${string}`>]: Match<T[K], GetMethod<any[], any>>
+} & {
+  [K in Keys<T, `delete${string}`>]: Match<T[K], DeleteMethod<any[]>>
+}
+/* eslint-enable @typescript-eslint/no-explicit-any */
 
-export class BaseRepository {
+export abstract class BaseRepository {
```

Comment:
Please, add a comment to this class explaining the conventions used here. Also, what exactly is the difference between findX and getX? This should be explained in the comment too! CC: @sz-piotr

Reply:
Sure, gonna do that in a minute. The difference is that get is returning list of records and find may return single record or undefined.

Reply:
That's what I thought... i was confused by the signature in CheckConvetion (`any[]` type). It's clear after reading `FindMethod` and `GetMethod` types 👍 But anyway, comment is needed.